extensions.html

```diff
@@ -209,33 +338,356 @@ <h2>
             values.
           </p>
         </dd>
-      </dl>
-    </section>
-    <section>
-      <h2>
-        <dfn>GamepadHapticActuatorType</dfn> Enum
-      </h2>
-      <p>
-        The actuator type determines the force applied for a given
-        <var>value</var> in <code>GamepadHapticActuator.pulse()</code>.
-      </p>
-      <pre class="idl">
-        enum GamepadHapticActuatorType {
-          "vibration"
-        };
-      </pre>
-      <dl data-dfn-for="GamepadHapticActuatorType">
         <dt>
-          <dfn>vibration</dfn>
+          <dfn>reset()</dfn>
         </dt>
         <dd>
-          Vibration is a rumbling effect often implemented as an offset weight
-          driven on a rotational axis. The <code>value</code> of a vibration
-          force determines the frequency of the rumble effect and is normalized
-          between <code>0.0</code> and <code>1.0</code>. The neutral value is
-          <code>0.0</code>.
+          <p>
+            The {{GamepadHapticActuator/reset()}} method steps are:
+          </p>
+          <ol>
+            <li>Let |document| be the [=current settings object=]'s [=relevant
+            global object=]'s [=associated `Document`=].
+            </li>
+            <li>If |document| is `null` or |document| is not [=Document/fully
+            active=] or |document|'s [=visibility state=] is `"hidden"`, return
+            [=a promise resolved with=] {{GamepadHapticsResult/"preempted"}}.
+            </li>
+            <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            is not `undefined`, [=queue a global task=] on the [=relevant
+            global object=] of [=this=] using the [=gamepad task source=] to
+            run the following steps:
+              <ol>
+                <li>[=Resolve=]
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+                with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `undefined`.
+                </li>
+              </ol>
+            </li>
+            <li>[=Stop haptic effects=] on this actuator.
+            </li>
+            <li>Return [=a promise resolved with=]
+            {{GamepadHapticsResult/"complete"}};
+            </li>
+          </ol>
+        </dd>
+        <dt>
+          Construction
+        </dt>
+        <dd>
+          <p>
+            Add new steps to [=a new `Gamepad`=]:
+          </p>
+          <ol>
+            <li>If |gamepad:Gamepad| has a vibration actuator that can [=play
+            effects with type=] {{GamepadHapticEffectType/"dual-rumble"}},
+            initialize |gamepad:Gamepad|'s {{Gamepad/vibrationActuator}}
+            attribute to [=a new `GamepadHapticActuator`=] with
+            {{GamepadHapticActuatorType/"dual-rumble"}}.
```

Comment:
Hmm.. I'm a bit confused... the `GamepadHapticActuator` seems to be both an actuator "hub" but also a place play the effects. This doesn't feel right. 


Shouldn't it be:

```
Gamepad 
   -> actuator
        ->  "dual-rumble" === "play effect on handles"
        ->  "xxxxx-rumble" === "play xxxxx on some part of the controller"
```

What's confusing is the `.type` attribute. What does it mean for `.type` to be "dual-rumble" if I can play "trigger-rumble"?

Reply:
I think we (Chromium) should deprecate and remove the actuator's `type` attribute since it's not useful for capability detection. Going forward, all vibration actuators that support "dual-rumble" effects should use type "dual-rumble" even if they support other effects. Once we've provided an alternate means of capability detection (either `canPlay` or `actuator.effects.includes`) then we can remove the `type` attribute.

In general, I don't want compatibility to be limited by the actuator type. It should be possible for user agents to emulate effects even when the actuator doesn't natively support that type of effect. For instance, a Switch Pro controller with an LRA in each handle might be able to play a "dual-rumble" effect meant for ERM actuators if the user agent knows how to simulate a dual-rumble effect with LRAs.

I agree `GamepadHapticActuator` is confusing. In my view it doesn't represent a specific physical actuator on the gamepad, it represents a system of actuators that can work together to create vibration effects. If a `GamepadHapticActuator` supports an effect type, that means its set of actuators includes the appropriate actuators to simulate the effect.

I originally thought that the `type` would help applications understand which effect type is preferred by the actuator. For example, a user agent might allow a "vibration" actuator to play "dual-rumble" effects as a fallback, with some loss in fidelity. The `type` attribute would let applications know to prefer "vibration" effects over "dual-rumble". I've changed my mind and I don't think this is useful anymore.